src/modules/ethereum/chains/ChainConfigStore.ts

```diff
@@ -38,27 +19,49 @@ function remove<T>(arr: T[], predicate: (item: T) => boolean) {
 
 class ChainConfigStore extends PersistentStore<ChainConfig> {
   static initialState: ChainConfig = {
+    version: 2,
     ethereumChainConfigs: [],
-    migratedChainConfigs: false,
-    migratedPermissions: false,
   };
 
+  async restore() {
+    const saved = await PersistentStore.readSavedState<ChainConfig>(this.key);
+    if (saved) {
+      this.setState(upgradeRecord(saved, upgrades));
+    }
+    this.isReady = true;
+    this.checkChainsForUpdates();
+  }
+
+  readyPromise = this.restore();
+
   addEthereumChain(
     value: AddEthereumChainParameter,
-    { origin, id, created }: { origin: string; id: string; created?: number }
+    {
+      origin,
+      id,
+      created,
+      prevId: maybePrevId,
+    }: { origin: string; id: string; created?: number; prevId?: string }
   ): EthereumChainConfig {
+    const prevId = maybePrevId || id;
     const state = this.getState();
     const existingItems = new Map(
       state.ethereumChainConfigs?.map((config) => [config.id, config])
     );
-    const existingEntry = existingItems.get(id);
+    const existingEntry = existingItems.get(prevId);
+    const existingPreviousIds = existingEntry?.previousIds || [];
+    const previousIds =
+      prevId !== id && !existingPreviousIds.includes(prevId)
+        ? [...existingPreviousIds, prevId]
```

Comment:
Why do we need to save this?

Reply:
Cause id can be changed for custom networks. And I want to save all history in this case.
However, these aliases will be less in priority, so should work almost fine

Reply:
Yes, there can be a case when we updating id to custom and then we fetched the info that backend supports this chain. So there should be 2 prev ids in this case.